packages/ui/src/primitives/field.tsx

```diff
@@ -40,31 +48,73 @@ export const Label = React.forwardRef<HTMLLabelElement, React.HTMLAttributes<HTM
 
 export const Input = React.forwardRef(function Input(
   {
+    asChild,
     className,
     intent = 'idle',
+    state = 'native',
     ...props
-  }: React.InputHTMLAttributes<HTMLInputElement> & {
+  }: Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'> & {
+    asChild?: boolean;
     intent?: FieldIntent;
+    type?: React.InputHTMLAttributes<HTMLInputElement>['type'] | 'otp';
+    state?: 'native' | 'hover' | 'focus-visible';
   },
   forwardedRef: React.ForwardedRef<HTMLInputElement>,
 ) {
+  const Comp = asChild ? Slot : 'input';
+
   return (
-    <input
+    <Comp
       data-field-input=''
       ref={forwardedRef}
       className={cn(
-        'block w-full bg-white text-gray-12 rounded-md bg-clip-padding py-1.5 px-2.5 border border-gray-a6 outline-none text-base',
-        'focus-visible:ring-[0.1875rem] disabled:opacity-50 disabled:cursor-not-allowed',
+        'flex w-full min-w-0 bg-white text-gray-12 rounded-md bg-clip-padding py-1.5 px-2.5 border outline-none text-base',
+        'disabled:opacity-50 disabled:cursor-not-allowed',
+        // type-specific overrides
+        props.type === 'otp'
+          ? 'min-h-10 text-[calc(var(--cl-font-size)*1.4)] font-semibold justify-center'
+          : 'min-h-8 justify-start',
```

Comment:
🤔 hm, mixed feelings on adding type specific logic like this within the input component. Since we are forwarding and input component onto a span. We need to make sure no input specific props are added in here to avoid invalid markup being added where used. Lets move forward as is for now.

Reply:
I guess my thinking here is that I didn't want to create some arbitrary abstraction that could be misused, but totally get you on the props part. Wondering if there's a way to circumvent this prop application behaviour somehow?